### PR TITLE
switch from deprecated 'header' hook to 'displayHeader'

### PR DESCRIPTION
**Pull request by tswfi**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Description?  | 'header' hook is deprecated, use 'displayHeader' instead
| Type?         | improvement
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | Fixes PrestaShop/Prestashop#27129 
| How to test?  | Check that the module works and add to basket works after upgrading the module

Not sure if the version number should be increased in a PR like this or should there be another pr for it..

### CODE REVIEW

File: config.xml [hunk 1]
```diff
@@ -2,7 +2,7 @@
 <module>
     <name>productcomments</name>
     <displayName><![CDATA[Product Comments]]></displayName>
-    <version><![CDATA[5.0.1]]></version>
+    <version><![CDATA[5.0.2]]></version>
```
Comment (PierreRambaud) [inline config.xml:5]:
```suggestion
    <version><![CDATA[5.0.1]]></version>
```

File: productcomments.php [hunk 2]
```diff
@@ -45,7 +45,7 @@ public function __construct()
     {
         $this->name = 'productcomments';
         $this->tab = 'front_office_features';
-        $this->version = '5.0.1';
+        $this->version = '5.0.2';
```
Comment (PierreRambaud) [inline productcomments.php:48]:
```suggestion
        $this->version = '5.0.1';
```

**Review by PierreRambaud — CHANGES_REQUESTED** [review]
The version 5.0.1 hasn't been released yet

**Comment by tswfi** [comment]
> The version 5.0.1 hasn't been released yet

Ah, so it seems. I can add the hook upgrade there.

**Comment by tswfi** [comment]
btw, is there a best practice on registerhook / unregisterhook. which one should be done first?

**Review by PierreRambaud — APPROVED** [review]

**Comment by PierreRambaud** [comment]
> btw, is there a best practice on registerhook / unregisterhook. which one should be done first?

Nope, there is nothing related to it ^^

**Comment by PierreRambaud** [comment]
Thank you @tswfi

**Pull request merged**